fhir-search/src/test/java/com/ibm/fhir/search/test/TotalParameterParseTest.java

```diff
@@ -0,0 +1,105 @@
+ * SearchUtil class.
+ *
+ * @author tbieste
+ *
+ */
+public class TotalParameterParseTest extends BaseSearchTest {
+
+    @Test
+    public void testTotalNotSpecified() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        FHIRSearchContext context = SearchUtil.parseQueryParameters(Patient.class, queryParameters);
+        assertNotNull(context);
+        assertNull(context.getTotalParameter());
+    }
+
+    @Test
+    public void testTotal() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        queryParameters.put("_total", Arrays.asList("estimate"));
+        FHIRSearchContext context = SearchUtil.parseQueryParameters(Patient.class, queryParameters);
+        assertNotNull(context);
+        assertEquals(context.getTotalParameter(), TotalValueSet.ESTIMATE);
+    }
+
+    @Test
+    public void testTotalInvalid_lenient() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        queryParameters.put("_total", Arrays.asList("invalid"));
+        FHIRSearchContext context = SearchUtil.parseQueryParameters(Patient.class, queryParameters, true);
+        assertNotNull(context);
+        assertNull(context.getTotalParameter());
+    }
+
+    @Test
+    public void testTotalInvalid_strict() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        boolean isExceptionThrown = false;
+
+        queryParameters.put("_total", Arrays.asList("invalid"));
+        try {
+            SearchUtil.parseQueryParameters(Patient.class, queryParameters, false);
+        } catch(Exception ex) {
+            isExceptionThrown = true;
+            assertEquals(ex.getMessage(), "An error occurred while parsing parameter '_total'.");
+
+        }
+        assertTrue(isExceptionThrown);
+    }
+
+    @Test
+    public void testTotalMultipleParams_lenient() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        queryParameters.put("_total", Arrays.asList("none", "accurate"));
+        FHIRSearchContext context = SearchUtil.parseQueryParameters(Patient.class, queryParameters, true);
+        assertNotNull(context);
+        assertEquals(context.getTotalParameter(), TotalValueSet.NONE);
+    }
+
+    @Test
+    public void testTotalMultipleParams_strict() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        boolean isExceptionThrown = false;
+
+        queryParameters.put("_total", Arrays.asList("none", "accurate"));
+        try {
+            SearchUtil.parseQueryParameters(Patient.class, queryParameters, false);
+        } catch(Exception ex) {
+            isExceptionThrown = true;
+            assertEquals(ex.getMessage(), "Search parameter '_total' is specified multiple times");
+
+        }
+        assertTrue(isExceptionThrown);
```

Comment:
Similar to SummaryParameterParseTest.java, I wanted to verify that the error message text was also correct, which I would lose the ability to do by using the expectedExceptions parameter of the @Test annotation.  I could do that, but felt it would be better to be consistent with the existing testcases.